### ISSUE TITLE
After running the server with 'rails s', when I post with curl I get an error: 'Couldn't find Site without an ID'

### ISSUE THREAD

**Issue by christophernies** [GreenSTEM-Network/GreenSTEM-Server#3]
[nies@localhost Server]$ curl -v -H "Content-Type: application/json" -X POST -d '{"type":"soilmoisture", "data":{"collection_point_id":1,"collection_time":"2013-03-08 08:58AM","deptha":1,"depthb":1.2,"depthc":1.9}}' 'http://0.0.0.0:3000/dc/'
- About to connect() to 0.0.0.0 port 3000 (#0)
-   Trying 0.0.0.0...
- Connected to 0.0.0.0 (0.0.0.0) port 3000 (#0)
  > POST /dc/ HTTP/1.1
  > User-Agent: curl/7.29.0
  > Host: 0.0.0.0:3000
  > Accept: _/_
  > Content-Type: application/json
  > Content-Length: 133
  > 
- upload completely sent off: 133 out of 133 bytes
  < HTTP/1.1 500 Internal Server Error
  < Content-Type: text/plain; charset=utf-8
  < X-Request-Id: 1a6b35cdf7355942ad5924902eeb0a22
  < X-Runtime: 0.048304
  < Connection: close
  < Server: thin 1.5.1 codename Straight Razor
  < 
  ActiveRecord::RecordNotFound at /dc
  ===================================

> Couldn't find Site without an ID
## app/controllers/data_collection_controller.rb, line 6

``` ruby
    1   class DataCollectionController < ApplicationController
    2   
    3       skip_before_filter :verify_authenticity_token
    4   
    5       def submit
>   6         site = Site.find(params[:site_id])
    7   
    8         DataCollectionContext.new(site, params).collect!
    9   
   10         render :json => 'success'
   11   
```
## App backtrace
- app/controllers/data_collection_controller.rb:6:in `submit'
## Full backtrace
- (gem) activerecord-3.2.12/lib/active_record/relation/finder_methods.rb:310:in `find_with_ids'
- (gem) activerecord-3.2.12/lib/active_record/relation/finder_methods.rb:107:in `find'
- (gem) activerecord-3.2.12/lib/active_record/querying.rb:5:in `find'
- app/controllers/data_collection_controller.rb:6:in `submit'
- (gem) actionpack-3.2.12/lib/action_controller/metal/implicit_render.rb:4:in `send_action'
- (gem) actionpack-3.2.12/lib/abstract_controller/base.rb:167:in `process_action'
- (gem) actionpack-3.2.12/lib/action_controller/metal/rendering.rb:10:in `process_action'
- (gem) actionpack-3.2.12/lib/abstract_controller/callbacks.rb:18:in `block in process_action'
- (gem) activesupport-3.2.12/lib/active_support/callbacks.rb:403:in `_run__4581888185925427691__process_action__2736955347581762706__callbacks'
- (gem) activesupport-3.2.12/lib/active_support/callbacks.rb:405:in `__run_callback'
- (gem) activesupport-3.2.12/lib/active_support/callbacks.rb:385:in `_run_process_action_callbacks'
- (gem) activesupport-3.2.12/lib/active_support/callbacks.rb:81:in `run_callbacks'
- (gem) actionpack-3.2.12/lib/abstract_controller/callbacks.rb:17:in `process_action'
- (gem) actionpack-3.2.12/lib/action_controller/metal/rescue.rb:29:in `process_action'
- (gem) actionpack-3.2.12/lib/action_controller/metal/instrumentation.rb:30:in `block in process_action'
- (gem) activesupport-3.2.12/lib/active_support/notifications.rb:123:in `block in instrument'
- (gem) activesupport-3.2.12/lib/active_support/notifications/instrumenter.rb:20:in `instrument'
- (gem) activesupport-3.2.12/lib/active_support/notifications.rb:123:in `instrument'
- (gem) actionpack-3.2.12/lib/action_controller/metal/instrumentation.rb:29:in `process_action'
- (gem) actionpack-3.2.12/lib/action_controller/metal/params_wrapper.rb:207:in `process_action'
- (gem) activerecord-3.2.12/lib/active_record/railties/controller_runtime.rb:18:in `process_action'
- (gem) actionpack-3.2.12/lib/abstract_controller/base.rb:121:in `process'
- (gem) actionpack-3.2.12/lib/abstract_controller/rendering.rb:45:in `process'
- (gem) actionpack-3.2.12/lib/action_controller/metal.rb:203:in `dispatch'
- (gem) actionpack-3.2.12/lib/action_controller/metal/rack_delegation.rb:14:in `dispatch'
- (gem) actionpack-3.2.12/lib/action_controller/metal.rb:246:in `block in action'
- (gem) actionpack-3.2.12/lib/action_dispatch/routing/route_set.rb:73:in `call'
- (gem) actionpack-3.2.12/lib/action_dispatch/routing/route_set.rb:73:in `dispatch'
- (gem) actionpack-3.2.12/lib/action_dispatch/routing/route_set.rb:36:in `call'
- (gem) journey-1.0.4/lib/journey/router.rb:68:in `block in call'
- (gem) journey-1.0.4/lib/journey/router.rb:56:in `each'
- (gem) journey-1.0.4/lib/journey/router.rb:56:in `call'
- (gem) actionpack-3.2.12/lib/action_dispatch/routing/route_set.rb:601:in `call'
- (gem) rack-pjax-0.7.0/lib/rack/pjax.rb:12:in `call'
- (gem) warden-1.2.1/lib/warden/manager.rb:35:in `block in call'
- (gem) warden-1.2.1/lib/warden/manager.rb:34:in `catch'
- (gem) warden-1.2.1/lib/warden/manager.rb:34:in `call'
- (gem) actionpack-3.2.12/lib/action_dispatch/middleware/best_standards_support.rb:17:in `call'
- (gem) rack-1.4.5/lib/rack/etag.rb:23:in `call'
- (gem) rack-1.4.5/lib/rack/conditionalget.rb:35:in `call'
- (gem) actionpack-3.2.12/lib/action_dispatch/middleware/head.rb:14:in `call'
- (gem) remotipart-1.0.2/lib/remotipart/middleware.rb:30:in `call'
- (gem) actionpack-3.2.12/lib/action_dispatch/middleware/params_parser.rb:21:in `call'
- (gem) actionpack-3.2.12/lib/action_dispatch/middleware/flash.rb:242:in `call'
- (gem) rack-1.4.5/lib/rack/session/abstract/id.rb:210:in `context'
- (gem) rack-1.4.5/lib/rack/session/abstract/id.rb:205:in `call'
- (gem) actionpack-3.2.12/lib/action_dispatch/middleware/cookies.rb:341:in `call'
- (gem) activerecord-3.2.12/lib/active_record/query_cache.rb:64:in `call'
- (gem) activerecord-3.2.12/lib/active_record/connection_adapters/abstract/connection_pool.rb:479:in `call'
- (gem) actionpack-3.2.12/lib/action_dispatch/middleware/callbacks.rb:28:in `block in call'
- (gem) activesupport-3.2.12/lib/active_support/callbacks.rb:405:in `_run__3666827810107932380__call__1724586944785535493__callbacks'
- (gem) activesupport-3.2.12/lib/active_support/callbacks.rb:405:in `__run_callback'
- (gem) activesupport-3.2.12/lib/active_support/callbacks.rb:385:in `_run_call_callbacks'
- (gem) activesupport-3.2.12/lib/active_support/callbacks.rb:81:in `run_callbacks'
- (gem) actionpack-3.2.12/lib/action_dispatch/middleware/callbacks.rb:27:in `call'
- (gem) actionpack-3.2.12/lib/action_dispatch/middleware/reloader.rb:65:in `call'
- (gem) actionpack-3.2.12/lib/action_dispatch/middleware/remote_ip.rb:31:in `call'
- (gem) better_errors-0.6.0/lib/better_errors/middleware.rb:71:in `protected_app_call'
- (gem) better_errors-0.6.0/lib/better_errors/middleware.rb:66:in `better_errors_call'
- (gem) better_errors-0.6.0/lib/better_errors/middleware.rb:41:in `call'
- (gem) actionpack-3.2.12/lib/action_dispatch/middleware/debug_exceptions.rb:16:in `call'
- (gem) actionpack-3.2.12/lib/action_dispatch/middleware/show_exceptions.rb:56:in `call'
- (gem) railties-3.2.12/lib/rails/rack/logger.rb:32:in `call_app'
- (gem) railties-3.2.12/lib/rails/rack/logger.rb:16:in `block in call'
- (gem) activesupport-3.2.12/lib/active_support/tagged_logging.rb:22:in `tagged'
- (gem) railties-3.2.12/lib/rails/rack/logger.rb:16:in `call'
- (gem) quiet_assets-1.0.1/lib/quiet_assets.rb:20:in `call_with_quiet_assets'
- (gem) actionpack-3.2.12/lib/action_dispatch/middleware/request_id.rb:22:in `call'
- (gem) rack-1.4.5/lib/rack/methodoverride.rb:21:in `call'
- (gem) rack-1.4.5/lib/rack/runtime.rb:17:in `call'
- (gem) activesupport-3.2.12/lib/active_support/cache/strategy/local_cache.rb:72:in `call'
- (gem) rack-1.4.5/lib/rack/lock.rb:15:in `call'
- (gem) actionpack-3.2.12/lib/action_dispatch/middleware/static.rb:62:in `call'
- (gem) railties-3.2.12/lib/rails/engine.rb:479:in `call'
- (gem) railties-3.2.12/lib/rails/application.rb:223:in `call'
- (gem) rack-1.4.5/lib/rack/content_length.rb:14:in `call'
- (gem) railties-3.2.12/lib/rails/rack/log_tailer.rb:17:in `call'
- (gem) thin-1.5.1/lib/thin/connection.rb:81:in `block in pre_process'
- (gem) thin-1.5.1/lib/thin/connection.rb:79:in `catch'
- (gem) thin-1.5.1/lib/thin/connection.rb:79:in `pre_process'
- (gem) thin-1.5.1/lib/thin/connection.rb:54:in `process'
- (gem) thin-1.5.1/lib/thin/connection.rb:39:in `receive_data'
- (gem) eventmachine-1.0.3/lib/eventmachine.rb:187:in `run_machine'
- (gem) eventmachine-1.0.3/lib/eventmachine.rb:187:in `run'
- (gem) thin-1.5.1/lib/thin/backends/base.rb:63:in `start'
- (gem) thin-1.5.1/lib/thin/server.rb:159:in `start'
- (gem) rack-1.4.5/lib/rack/handler/thin.rb:13:in `run'
- (gem) rack-1.4.5/lib/rack/server.rb:268:in `start'
- (gem) railties-3.2.12/lib/rails/commands/server.rb:70:in `start'
- (gem) railties-3.2.12/lib/rails/commands.rb:55:in `block in <top (required)>'
- (gem) railties-3.2.12/lib/rails/commands.rb:50:in `tap'
- (gem) railties-3.2.12/lib/rails/commands.rb:50:in `<top (required)>'
- script/rails:6:in `require'
- script/rails:6:in `<main>'
- Closing connection 0

**Comment by jasonblanchard** [comment]
See https://github.com/SolarSunflowerProject/Server/issues/2#issuecomment-30084338.

SInce `rake db:seed` didn't complete, there aren't any site or node IDs. 

Also, that's the wrong curl request. It should be like this:

curl -v -H "Content-Type: application/json" -X POST -d '{"site_id":"1","node_readings":[{"id":"1","timestamp":"2013-12-07 04:05:06","channel":"001","soil1":"1.2","soil2":"1.3","soil3":"1.4","temp":"58","voltage":"1.4"}]}' 'http://localhost:3000/dc/'

(I thought I fixed this in the README, but apparently now. Will update now)

**Comment by christophernies** [comment]
Awesome - after clearing up #2, this is no longer an issue.
